message/infrastructure/http/subscriber.go

```diff
@@ -1,23 +1,69 @@
 package http
 
 import (
+	"context"
+	"encoding/json"
+	"io/ioutil"
+	"net"
 	"net/http"
+	"strings"
 	"sync"
 
+	"github.com/pkg/errors"
+
 	"github.com/ThreeDotsLabs/watermill"
 	"github.com/ThreeDotsLabs/watermill/message"
 	"github.com/go-chi/chi"
 )
 
 type UnmarshalMessageFunc func(topic string, request *http.Request) (*message.Message, error)
 
+// DefaultUnmarshalMessageFunc retrieves the UUID and Metadata from request headers,
+// as encoded by DefaultMarshalMessageFunc.
+func DefaultUnmarshalMessageFunc(topic string, req *http.Request) (*message.Message, error) {
+	body, err := ioutil.ReadAll(req.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	uuid := req.Header.Get(HeaderUUID)
+	if uuid == "" {
```

Comment:
it's imo fine, uuid is not required anywhere

Reply:
and it may be hard to enforce every client remote publisher to send this header (for example: gitlab)

Reply:
744769805d87476f08dc58de6626dfc38ae98125